src/Dfe.PlanTech.Application/Submissions/Queries/SectionCompleteStatusChecker.cs

```diff
@@ -0,0 +1,24 @@
+using Dfe.PlanTech.Domain.Submissions.Enums;
+using Dfe.PlanTech.Domain.Submissions.Interfaces;
+using Dfe.PlanTech.Domain.Submissions.Models;
+
+namespace Dfe.PlanTech.Application.Submissions.Queries;
+
+/// <summary>
+/// User journey status checker for when the section for the establishment is complete
+/// </summary>
+public static class SectionCompleteStatusChecker
+{
+  public static readonly ISubmissionStatusChecker SectionComplete = new SubmissionStatusChecker()
+  {
+    IsMatchingSubmissionStatusFunc = (userJourneyRouter) => userJourneyRouter?.SectionStatus?.Completed == true,
+    ProcessSubmissionFunc = (userJourneyRouter, cancellationToken) =>
+    {
+      userJourneyRouter.Status = SubmissionStatus.Completed;
+      userJourneyRouter.NextQuestion = userJourneyRouter.Section!.Questions[0];
```

Comment:
Could questions ever be null or have no items?

Reply:
Shouldn't be, as Questions are required from Contentful. 

However, they most certainly actually could be. In which case, I actually want this to error here. I've not caught it anywhere as the cause should be obvious from the exception thrown.